python/pdstools/app/datamart_utils.py

```diff
@@ -48,15 +43,21 @@ def import_data(params, default=0, **kwargs):
             )
             params["kwargs"] = ADMDatamart_options()
             if st.checkbox("Use this data source"):
-                if model_file is not None:
+                if predictor_file is None and model_file is not None:
+                    data = import_datamart(
+                        ADMDatamart,
+                        model_df=model_file,
+                        predictor_df=predictor_file,
```

Comment:
If you give `predictor_filename = None`, then please don't pass a `predictor_df`. filenames take priority. So, if you want to _not_ pass a predictordata at all, just pass `predictor_filename = None` and no `predictor_df`.